src/Contrib/Jaeger/ThriftHttpSender.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\Jaeger;
+
+use Jaeger\Thrift\Batch;
+use Jaeger\Thrift\Process;
+use Jaeger\Thrift\Span as JTSpan;
+use OpenTelemetry\SDK\Behavior\LogsMessagesTrait;
+use Thrift\Protocol\TBinaryProtocol;
+use Thrift\Protocol\TProtocol;
+use Thrift\Transport\THttpClient;
+
+class ThriftHttpSender
+{
+    use LogsMessagesTrait;
+
+    private string $serviceName;
+
+    private TProtocol $protocol;
+
+    public function __construct(
+        string $serviceName,
+        string $host,
+        int $port,
+        string $path,
+        string $scheme
+    ) {
+        $this->serviceName = $serviceName;
+
+        $transport = new THttpClient(
```

Comment:
Please don't use this client (see my comment)

Reply:
Yeah I agree current state is not gonna work having stepped back from it a bit now. I'm thinking maybe I'll try 2 things along the lines of your direction

1) Rewrite THttpClient::flush to use a PSR-18 HTTP client injected by consumers (like I gather the other exporters are already doing from checking a few)
2) Try and see if apache/thrift would be interested in moving this change upstream (or if there's maybe some v subtle reason fopen has to be used there?) since it doesn't look like that code has been touched much since 3-4 years ago

I'll go ahead and pop open an issue in their repo for 2) to gauge interest, then start on giving 1) a shot

Reply:
Thank you.
I believe the initial code-base is older than PSR-18.
FWIW there is also a [TCurlClient](https://github.com/apache/thrift/blob/master/lib/php/lib/Transport/TCurlClient.php), however there are no tests for both clients.

Reply:
Took a stab at it. You can see my current progress in this draft PR I made inside my own fork - https://github.com/Grunet/opentelemetry-php/pull/2/files Still some cleanup work to do but figured it was shareable

I wanted to start with a normal composed adapter for THttpClient but realized eventually all of its state was in protected variables, so I switched to just a subclass (since it seemed like I'd need one anyways) and injected the extras it would need (not sure this is the best approach, just the first that came to mind after the pivot)

I also cut out a lot of the aspects of the "flush" method that otel-php doesn't need to use, so probably no simple path to upstreaming this at the moment.

Would appreciate any pointers or comments on things that are obv off (this is the first time I think I've interacted with the PSR-18/PSR-7 interfaces directly, and it was a neat experience!)

Reply:
@Grunet 
Sorry for the late reply, I was a bit busy.
Let's talk about this in tomorrows meeting, if you still have questions.

Reply:
Went ahead and merged what I had into this branch. Lmk if the approach needs any adjustment (it's light on automated test coverage/assertions for the moment. Probably going to wait to see what doing the final adapting work for that commented out block of code in the next PR will do before adding some more)

Reply:
Ah yeah I see what you mean. It looks like the $batch->write($this->protocol) line only ends up calling into the "write" method of THttpClient, so I think I should probably be able to copy that and give stub implementations for the other TTransport interface methods I think.

Ok went ahead and gave this a try, seems like it went fine